src/Uno.UI.RemoteControl.VS/EntryPoint.ActiveProfileSync.cs

```diff
@@ -71,36 +71,40 @@ private async Task OnDebugFrameworkChangedAsync(string? previousFramework, strin
 
 			var profiles = await _debuggerObserver.GetLaunchProfilesAsync();
 
-			if (targetFrameworkIdentifier == WasmTargetFrameworkIdentifier)
+			var profileFilter = targetFrameworkIdentifier switch
 			{
-				if (profiles.Find(p => p.LaunchBrowser) is { } browserProfile)
-				{
-					_debugAction?.Invoke($"Setting profile {browserProfile.Name}");
+				WasmTargetFrameworkIdentifier
+					=> ((ILaunchProfile p) => p.LaunchBrowser),
 
-					await _debuggerObserver.SetActiveLaunchProfileAsync(browserProfile.Name);
-				}
-			}
-			else if (targetFrameworkIdentifier == DesktopTargetFrameworkIdentifier)
-			{
-				bool IsCompatible(ILaunchProfile profile)
-					=> profile.OtherSettings.TryGetValue(CompatibleTargetFrameworkProfileKey, out var compatibleTargetFramework)
-						&& compatibleTargetFramework is string ctfs
-						&& ctfs.Equals(DesktopTargetFrameworkIdentifier, StringComparison.OrdinalIgnoreCase);
+				DesktopTargetFrameworkIdentifier
+					=> (ILaunchProfile profile)
+						=> profile.OtherSettings.TryGetValue(CompatibleTargetFrameworkProfileKey, out var compatibleTargetFramework)
+							&& compatibleTargetFramework is string ctfs
+							&& ctfs.Equals(DesktopTargetFrameworkIdentifier, StringComparison.OrdinalIgnoreCase),
 
-				if (profiles.Find(IsCompatible) is { } desktopProfile)
-				{
-					_debugAction?.Invoke($"Setting profile {desktopProfile.Name}");
+				Windows10TargetFrameworkIdentifier
+					=> (ILaunchProfile p) => p.CommandName.Equals("MsixPackage", StringComparison.OrdinalIgnoreCase),
 
-					await _debuggerObserver.SetActiveLaunchProfileAsync(desktopProfile.Name);
-				}
-			}
-			else if (targetFrameworkIdentifier == Windows10TargetFrameworkIdentifier)
+				_ => (Func<ILaunchProfile, bool>?)null
+			};
+
+			if (profileFilter is not null)
```

Comment:
```suggestion
if (profileFilter != null)
{
    // If the current profile already matches the targetframework we're going to
    // prefer using it. It can happen if the change is initiated through the profile selector.
    var currentCompatible = profiles.FirstOrDefault(p => p.Name == _debuggerObserver.CurrentActiveDebugProfile && profileFilter(p));
    // Otherwise, select the first compatible profile.
    var firstCompatible = currentCompatible ?? profiles.FirstOrDefault(profileFilter);

    if (firstCompatible != null)
    {
        _debugAction?.Invoke($"Setting profile {firstCompatible}");
        await _debuggerObserver.SetActiveLaunchProfileAsync(firstCompatible.Name);
    }
}
```
Maybe we can streamline the evaluation of `currentCompatible` and `firstCompatible` with something like that?